osbuild/buildroot.py

```diff
@@ -263,3 +267,27 @@ def run(self, argv, monitor, binds=None, readonly_binds=None):
         data.close()
 
         return CompletedBuild(proc, output)
+
+    @classmethod
+    def read_with_timeout(cls, proc, poller, start, stage_timeout):
+        fd = proc.stdout.fileno()
+        if stage_timeout is None:
```

Comment:
Nitpick (could be fixed in follow-up): For the sake of uniformity, you could set `remaining=None` in case `stage_timeout is None`, in stead of calling `os.read()`.